src/Microsoft.DotNet.XHarness.CLI/Commands/Android/AndroidGetStateCommand.cs

```diff
@@ -25,24 +28,55 @@ public AndroidGetStateCommand() : base(TargetPlatform.Android, new ServiceCollec
 
     protected override Task<ExitCode> InvokeInternal(ILogger logger)
     {
-        var runner = new AdbRunner(logger);
-
-        logger.LogInformation("Getting state of ADB and attached Android device(s)");
         try
         {
-            string state = runner.GetAdbState();
-            if (string.IsNullOrEmpty(state))
+            var data = GetStateData(logger);
+
+            if (Arguments.UseJson)
             {
-                state = "No device attached";
-            }
-            logger.LogInformation($"ADB Version info:{Environment.NewLine}{runner.GetAdbVersion()}");
-            logger.LogInformation($"ADB State ('device' if physically attached):{Environment.NewLine}{state}");
+                var options = new JsonSerializerOptions
+                {
+                    WriteIndented = true
+                };
 
-            logger.LogInformation($"List of devices:");
-            var deviceAndArchList = runner.GetAttachedDevicesWithProperties("architecture");
-            foreach (string device in deviceAndArchList.Keys)
+                JsonSerializer.Serialize(Console.OpenStandardOutput(), data, options);
+            }
+            else

```

Comment:
In such long methods, it'd be more readable to return early instead of branching to else. The difference is that with returning early, I can immediately see that the method is done, with else block I have to scroll down to check if some logic follows yet. Also reduces indentation.

Reply:
Agreed, done